deep_qa/data/tokenizers/character_tokenizer.py

```diff
@@ -1,66 +1,39 @@
-from typing import Callable, Dict, List, Tuple
-from keras.layers import Layer
+from typing import List
 from overrides import overrides
 
-from .tokenizer import Tokenizer
-from ..data_indexer import DataIndexer
 from ...common.params import Params
+from .tokenizer import Tokenizer
 
 
 class CharacterTokenizer(Tokenizer):
     """
-    A CharacterTokenizer splits strings into character tokens.
-
-    Notes
-    -----
-    Note that in the code, we're still using the "words" namespace, and the "num_sentence_words"
-    padding key, instead of using a different "characters" namespace.  This is so that the rest of
-    the code doesn't have to change as much to just use this different tokenizer.  For example,
-    this is an issue when adding start and stop tokens - how is an ``Instance`` class supposed to
-    know if it should use the "words" or the "characters" namespace when getting a start token id?
-    If we just always use the "words" namespace for the top-level token namespace, it's not an
-    issue.
-
-    But confusingly, we'll still use the "characters" embedding key...  At least the user-facing
-    parts all use ``characters``; it's only in writing tokenizer code that you need to be careful
-    about namespaces.  TODO(matt): it probably makes sense to change the default namespace to
-    "tokens", and use that for both the words in ``WordTokenizer`` and the characters in
-    ``CharacterTokenizer``, so the naming isn't so confusing.
+    A ``CharacterTokenizer`` splits strings into character tokens.
+
+    Parameters
+    ----------
+    byte_encoding : str, optional (default=``None``)
+        If not ``None``, we will use this encoding to encode the string as bytes, and use the byte
+        sequence as characters, instead of the unicode characters in the python string.  E.g., the
+        character 'á' would be a single token if this option is ``None``, but it would be two
+        tokens if this option is set to ``"utf-8"``: there is one token for the accent and another
+        token for the 'a' character.  Note, though, that with a utf-8 encoding, the token for an
+        accented 'a' will be different than the token for an unaccented 'a' - hopefully your
+        embedding will decide that they are at least similar - while the token for the accent is
+        indeed shared across different accented vowels.
+    lowercase_characters : ``bool``, optional (default=``False``)
+        If ``True``, we will lowercase all of the characters in the text before doing any other
+        operation.  You probably do not want to do this, as character vocabularies are generally
+        not very large to begin with, but it's an option if you really want it.
     """
     def __init__(self, params: Params):
+        self.byte_encoding = params.pop('byte_encoding', None)
+        self.lowercase_characters = params.pop('lowercase_characters', False)
```

Comment:
That also means we would need a way to specify that `from_params`, which is possible, but more work.  I vote for switching that one later, when there's a compelling use for it.